### PR TITLE
SourceKit: libdispatch swift requires swift stdlib

### PR DESCRIPTION
**Pull request by compnerd**
When building libdispatch for SourceKit, only enable the swift overlay
components if the swift standard library is being built.  This allows building
just the compiler and SourceKit.

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by gottesmm** [comment]
@compnerd @akyrtzi Is this a useful build to verify? Maybe this needs a bot or something? Not sure.

File: tools/SourceKit/CMakeLists.txt [hunk 1]
```diff
@@ -93,6 +93,12 @@ include_directories(BEFORE
 if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
   set(SOURCEKIT_DEFAULT_TARGET_SDK "LINUX")
   if(SWIFT_BUILD_SOURCEKIT)
+    if(SWIFT_BUILD_STDLIB)
+      set(libdispatch_ENABLE_SWIFT YES)
```
Comment (akyrtzi) [inline tools/SourceKit/CMakeLists.txt]: Should `libdispatch_ENABLE_SWIFT` include the `SOURCEKIT` prefix ? It seems it has potential to get confused with some other global variable.

**Comment by akyrtzi** [comment]
> @compnerd @akyrtzi Is this a useful build to verify? Maybe this needs a bot or something? Not sure.

`SWIFT_BUILD_STDLIB` is not specific to SourceKit. If a bot is necessary or not it should be considered in the context of building the Swift compiler project as a whole with `SWIFT_BUILD_STDLIB` set to NO, SourceKit is just another piece of it.

File: tools/SourceKit/CMakeLists.txt (hunk 1 above)
Comment (compnerd) [reply tools/SourceKit/CMakeLists.txt]: Sure!  I'll adjust that.

**Comment by compnerd** [comment]
@swift-ci please test and merge

**Comment by compnerd** [comment]
@swift-ci please test and merge

**Pull request merged**